src/webhook_launcher/app/models.py

```diff
@@ -287,9 +288,15 @@ def revision(self):
 
     @property
     def lsr(self):
-        _lsr = self.lastseenrevision_set.all()
-        if _lsr:
-            return _lsr[0]
+        # TODO: refactor the WebHookMapping and LastSeenRevision relation
+        if not hasattr(self, '_lsr'):
+            if self.pk:
+                self._lsr, _ = LastSeenRevision.objects.get_or_create(
+                    mapping=self
+                )
+            else:
+                return None
+        return self._lsr
```

Comment:
Well it changes slightly. But I think it does not make any difference as the life cycle of a WHM object is not that long. 

There probably is chance for race conditions where one process is handling one webhook event when another comes in and ones changes get overwriten, but there haven't really been anything preventing that from happening earlier either... Except that if we run only one participant for handling the webhooks, then I think the events are queued.

This change was mainly to generalize the LastSeenRevision.objects.get_or_create(mapping=mapobj) that is used pretty much every where (which also shows that the FK relation from LSR to WHM is bit backwards, or it shoud be one2one relation)